File: Binary Search/binary_search.c
#include <stdio.h>
#include <stdbool.h>

int binary_search(long array[], int size, long value)
{
    int lower_bound = 0;
    int upper_bound = size - 1;
    int mid_point = -1;
    int index = -1;

    while (lower_bound <= upper_bound)
    {
        mid_point = lower_bound + (upper_bound - lower_bound) / 2;

        if (array[mid_point] == value)
        {
            index = mid_point;
            break;
        }
        else if (array[mid_point] < value)
            lower_bound = mid_point + 1;
        else if (array[mid_point] > value)
            upper_bound = mid_point - 1;

    }
    return index;
}


int main(void)
{
    long array[] = 
    {
       1, 2, 3, 4, 5, 6, 7, 8, 9, 10
    };


    int ret = binary_search(array, 10, 6);

    if (ret != -1)
        printf("Found value at index %d\n", ret);
    else
        printf("Failed to find value '%ld'\n", 6);

    return 0;
}

File: Binary Search/binary_search.py
def binary_search(arr, value):
    lower = 0
    upper = len(arr) - 1
    mid = -1
    index = -1

    while lower <= upper:
        mid = lower + (upper - lower) // 2
        if arr[mid] == value:
            index = mid
            break
        elif arr[mid] < value:
            lower = mid + 1
        elif arr[mid] > value:
            upper = mid - 1
    return index

l = [1, 2, 4, 6, 8, 9, 12, 13]

if binary_search(l, 9) > -1:
    print("Found value")
else:
    print("Failed to find value")